src/python_testing/TC_EWATERHTR_2_2.py

```diff
@@ -0,0 +1,376 @@
+                     "Verify value is 0x00 (no demand on any source)"),
+            TestStep("21b", "TH reads BoostState attribute.",
+                     "Verify value is 0 (Inactive)"),
+            TestStep("22", "TH sends CancelBoost.",
+                     "Verify Command response is Success"),
+            TestStep("23", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER for Basic installation Test Event Clear.",
+                     "Verify Command response is Success"),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EWATERHTR_2_2(self):
+
+        self.step("1")
+        # Commission DUT - already done
+
+        self.step("2")
+        await self.check_test_event_triggers_enabled()
+
+        self.step("3")
+        await self.send_test_event_trigger_basic_installation_test_event()
+
+        self.step("3a")
+        await self.check_whm_attribute("HeatDemand", 0)
+
+        self.step("3b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kInactive)
+
+        self.step("3c")
+        heaterTypes = await self.read_whm_attribute_expect_success(attribute="HeaterTypes")
+        asserts.assert_equal(
+            heaterTypes, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterTypeBitmap.kImmersionElement1)
+
+        self.step("4")
+        await self.send_test_event_trigger_manual_mode_test_event()
+
+        self.step("4a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
+            heatDemand & Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kImmersionElement1, 0)
+
+        self.step("5")
+        await self.send_test_event_trigger_water_temperature61C_test_event()
+
+        self.step("5a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_equal(heatDemand, 0)
+
+        self.step("6")
+        await self.send_test_event_trigger_water_temperature20C_test_event()
+
+        self.step("6a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
+            heatDemand & Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kImmersionElement1, 0)
+
+        self.step("7")
+        await self.send_test_event_trigger_off_mode_test_event()
+
+        self.step("7a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_equal(heatDemand, 0)
+
+        self.step("8")
+        await self.send_boost_command(duration=5, one_shot=True)
+
+        self.step("8a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
```

Comment:
Does not match test plan: 
{valhas} greater than 0x00 (demand on at least one source) and (HeaterDemand & (!HeaterTypes)) is zero (demand is only from declared supported types)

Reply:
Done